addons/survey/tests/test_survey_results.py

```diff
@@ -0,0 +1,244 @@
+        question answers data are stored in the same model.
+    """
+
+    def setUp(self):
+        self.SurveyController = Survey()
+
+        self.quiz_survey_id = self.ref('survey.survey_demo_quiz') # Quiz about our Company
+        self.quiz_survey = self.env['survey.survey'].browse(self.quiz_survey_id)
+
+        self.what_is_your_email = self.ref('survey.survey_demo_quiz_p1_q1') # Char box
+        self.mark_email_id = self.ref('survey.survey_demo_quiz_answer_1_p1_q1_l1')
+        self.mark_email_value = self.env['survey.user_input.line'].browse(self.mark_email_id).value_char_box
+
+        self.where_are_you_from_id = self.ref('survey.survey_demo_quiz_p1_q3') # Char box
+        self.brussels_id = self.ref('survey.survey_demo_quiz_answer_1_p1_q3_l1')
+        self.brussels_value = self.env['survey.user_input.line'].browse(self.brussels_id).value_char_box
+
+        self.citrus_could_give_you_id = self.ref('survey.survey_demo_quiz_p3_q4') # Multiple choice
+        self.pomelos_id = self.ref('survey.survey_demo_quiz_p3_q4_sug1')
+
+        self.which_use_to_pollinate_id = self.ref('survey.survey_demo_quiz_p3_q2') # Simple choice
+        self.bees_id = self.ref('survey.survey_demo_quiz_p3_q2_sug1')
+
+        self.how_often_should_you_water_id = self.ref('survey.survey_demo_quiz_p3_q5') # Matrix
+        self.cactus_row_id = self.ref('survey.survey_demo_quiz_p3_q5_row1')
+        self.once_a_month_id = self.ref('survey.survey_demo_quiz_p3_q5_sug1')
+
+        self.when_harvest_fruits_id = self.ref('survey.survey_demo_quiz_p3_q6') # Matrix
+        self.apple_row_id = self.ref('survey.survey_demo_quiz_p3_q6_row1')
+        self.spring_id = self.ref('survey.survey_demo_quiz_p3_q6_sug1')
+
+    def _check_results_and_query_count(self, post, lines_domain, expected_query_count):
+        """
+            Check that, depending on the URL filters, the _extract_filters_data method
+            is correctly returning the expected user input lines.
+            //As we cannot ref() all expected user input lines, we use
+            _get_matching_user_input_lines (tested below) to fetch them from the domain
+        """
+        with MockRequest(self.env):
+            with self.assertQueryCount(1):
+                base_domain = Survey._get_results_page_user_input_domain(self, self.quiz_survey, **post)
+                expected_user_input_line_ids = self.SurveyController._get_matching_user_input_lines(lines_domain, base_domain)
+
+            with self.assertQueryCount(expected_query_count):
+                found_user_input_line_ids, _ = self.SurveyController._extract_filters_data(self.quiz_survey, post)
+
+        self.assertEqual(expected_user_input_line_ids, found_user_input_line_ids)
+
+    def test_check_demo_question_type(self):
+        """
+            Making sure the demo questions used in this test class have the expected question types
+        """
```

Comment:
Emphasizing on the purpose of the test.

```suggestion
        """Check the assumptions on demo data, ensuring the tests' validity.
        """
```